internal/anchorplatform/object_schemas.go

```diff
@@ -78,39 +91,46 @@ type APSep24Transaction struct {
 	AmountOut *APAmount `json:"amount_out,omitempty"`
 	AmountFee *APAmount `json:"amount_fee,omitempty"`
 
+	// Accounts
+	SourceAccount      string `json:"source_account,omitempty"`
+	DestinationAccount string `json:"destination_account,omitempty"`
+
 	// Dates
 	CompletedAt        *time.Time `json:"completed_at,omitempty"`
 	TransferReceivedAt *time.Time `json:"transfer_received_at,omitempty"`
 }
 
-func NewAPSep24TransactionRecordsFromPatches(patches ...APSep24Transaction) APSep24TransactionRecords {
-	var records APSep24TransactionRecords
-	for _, patch := range patches {
-		newEntry := APSep24TransactionWrapper{
-			APSep24Transaction: patch,
-		}
-		records.Records = append(records.Records, newEntry)
-	}
-
-	return records
+// APSep24TransactionPatchPostRegistration is a subset of APSep24TransactionPatch that can be used to update the
+// transaction data after the registration.
+type APSep24TransactionPatchPostRegistration struct {
+	ID                    string              `json:"id"`
+	ExternalTransactionID string              `json:"external_transaction_id,omitempty"`
+	SEP                   string              `json:"sep,omitempty"`
+	Status                APTransactionStatus `json:"status,omitempty"`
+	Message               string              `json:"message,omitempty"`
+	TransferReceivedAt    *time.Time          `json:"transfer_received_at,omitempty"`
 }
 
-// type APSep24TransactionPostSuccess struct {
+// type APSep24TransactionPatchPostSuccess struct {
 // 	ID                  string                 `json:"id"`
 // 	SEP                 string                 `json:"sep,omitempty"`
 // 	Status              APTransactionStatus    `json:"status,omitempty"` // Success
 // 	StellarTransactions []APStellarTransaction `json:"stellar_transactions,omitempty"`
 // 	Message             string                 `json:"message,omitempty"`
 // 	CompletedAt         *time.Time             `json:"completed_at,omitempty"`
 // 	AmountOut           APAmount               `json:"amount_out,omitempty"`
+// 	// TODO: update the AP version when source_account becomes patchable
+// 	SourceAccount string `json:"source_account,omitempty"`
 // }
 
-// type APSep24TransactionPostError struct {
+// type APSep24TransactionPatchPostError struct {
 // 	ID                  string                 `json:"id"`
 // 	SEP                 string                 `json:"sep,omitempty"`
-// 	Status              APTransactionStatus    `json:"status,omitempty"` // Error
 // 	StellarTransactions []APStellarTransaction `json:"stellar_transactions,omitempty"`
 // 	Message             string                 `json:"message,omitempty"` // Error message
+// 	Status              APTransactionStatus    `json:"status,omitempty"`  // Error
+// 	// TODO: update the AP version when source_account becomes patchable
+// 	SourceAccount string `json:"source_account,omitempty"`
 // }
```

Comment:
These objects were created to make it simple for us to know which fields we will want to update on case of Registration, PaymentSuccess of PaymentFailure.